index.js

```diff
@@ -408,111 +413,121 @@ function Jasmine2ScreenShotReporter(opts) {
     };
 
     this.specDone = function(spec) {
-        var file;
+        var file, loopInstances;
         spec = getSpecClone(spec);
         spec._finished = Date.now();
 
         if (!isSpecValid(spec)) {
-          spec.skipPrinting = true;
-          return;
+            spec.skipPrinting = true;
+            return;
         }
 
-        browser.takeScreenshot().then(function (png) {
-            browser.getCapabilities().then(function (capabilities) {
-                var screenshotPath,
-                    metadataPath,
-                    metadata;
-
-                file = opts.pathBuilder(spec, suites, capabilities);
-                spec.filename = file + '.png';
-
-                screenshotPath = path.join(opts.dest, spec.filename);
-                metadata       = opts.metadataBuilder(spec, suites, capabilities);
-
-                if (metadata) {
-                    metadataPath = path.join(opts.dest, file + '.json');
-                    mkdirp(path.dirname(metadataPath), function(err) {
-                        if(err) {
-                            throw new Error('Could not create directory for ' + metadataPath);
+        loopInstances = protractor.hasOwnProperty('myInstances') ? protractor.myInstances : { 'screenshot': browser };
+        spec.filename = {};
+        for (var key in loopInstances) {
+            (function(key) {
```

Comment:
there's no need for this for-loop-closure magic, there's lodash available in the project -> you can just `_.each(loopInstances, ...)`